cis_proc.py

```diff
@@ -162,209 +162,58 @@ def main(tar_file, bids_dir, config, sub, ses=None, work_dir=None, n_procs=1):
-            if op.isfile(task_json_file):
-                run_mriqc = True
-
-            if run_mriqc:
-                kwarg_str = ''
-                settings_dict = mriqc_tasks[tmp_task]['mriqc_settings']
-                for field in settings_dict.keys():
-                    if isinstance(settings_dict[field], list):
-                        val = ' '.join(settings_dict[field])
-                    else:
-                        val = settings_dict[field]
-                    kwarg_str += '--{0} {1} '.format(field, val)
-                kwarg_str = kwarg_str.rstrip()
-                cmd = ('singularity run --cleanenv '
-                       '-B {templateflowdir}:$HOME/.cache/templateflow '
-                       '{sing} {bids} {out} participant '
-                       '--no-sub --verbose-reports '
-                       '--task-id {task} -m bold '
-                       '-w {work} --n_procs {n_procs} --correct-slice-timing '
-                       '{kwarg_str}'.format(
-                           templateflowdir=op.join(work_dir, 'templateflow'),
-                           sing=scratch_mriqc,
-                           bids=scratch_bids_dir,
-                           out=scratch_deriv_dir,
-                           task=tmp_task,
-                           work=mriqc_work_dir,
-                           n_procs=n_procs,
-                           kwarg_str=kwarg_str))
-                run(cmd)
-
-        # Merge MRIQC results into final derivatives folder
-        reports = glob(op.join(scratch_deriv_dir, '*.html'))
-        reports = [f for f in reports if 'group_' not in op.basename(f)]
-        for report in reports:
-            shutil.copy(report, op.join(out_deriv_dir, op.basename(report)))
-
-        logs = glob(op.join(scratch_deriv_dir, 'logs/*'))
-        for log in logs:
-            shutil.copy(log, op.join(out_deriv_dir, 'logs', op.basename(log)))
-
-        derivatives = glob(op.join(scratch_deriv_dir, 'sub-*'))
-        derivatives = [x for x in derivatives if '.html' not in op.basename(x)]
-        for derivative in derivatives:
-            shutil.copytree(
-                derivative,
-                op.join(out_deriv_dir, op.basename(derivative))
-            )
-
-        csv_files = glob(op.join(scratch_deriv_dir, '*.csv'))
-        for csv_file in csv_files:
-            out_file = op.join(out_deriv_dir, op.basename(csv_file))
-            if not op.isfile(out_file):
-                shutil.copyfile(csv_file, out_file)
-            else:
-                new_df = pd.read_csv(csv_file)
-                old_df = pd.read_csv(out_file)
-                out_df = pd.concat((old_df, new_df))
-                out_df.to_csv(out_file, index=False)
-
-        # Finally, clean up working directory *if successful*
-        shutil.rmtree(scan_work_dir)
-    else:
+    if not bids_successful:
         print('Heudiconv-generated dataset failed BIDS validator. '
               'Not running MRIQC')
+        return
```

Comment:
Is there a reason that this is a return and not a raise? Anything that passes `cis-bidsify` should be BIDS Valid barring some major issues.

Reply:
It was always a return, so I didn't put much thought into it in the refactor. I'm honestly not sure why we have a return instead of an exception.